### PR TITLE
New Rule: disallowTabs

### PR DESCRIPTION
**Pull request by ottumm**
Disallows tabs anywhere in the file/input.

### CODE REVIEW

File: lib/rules/disallow-tabs.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    },
+
+    getOptionName: function() {
+        return 'disallowTabs';
+    },
+
+    check: function(file, errors) {
+        file.getLines().forEach(function(line, i) {
+            if (line.match(/\t/)) {
+                errors.add('Tab found', i + 1, 0);
```
Comment (markelog) [inline lib/rules/disallow-tabs.js]: Maybe we could be more specific then that?

**Comment by markelog** [comment]
I guess this duplicates in some way duplicates
```js
{ 
  "validateIndentation": 4
}
```

But not fully

**Comment by mikesherov** [comment]
@ottumm thanks for contributing? what about tabs within strings or comments?

**Comment by ottumm** [comment]
@markelog yeah, would it be better to extend validateIndentation to handle this? My use case is to completely disallow tabs. I was finding that validateIndentation wasn't doing that and in fact was exhibiting problems when tabs were present. 

@mikesherov you're referring to literal tabs within strings, right (not '\t')? I was under the impression that tabs need to be escaped in JavaScript strings. 

Regarding comments, I could add an option to ignore tabs within comments.

File: lib/rules/disallow-tabs.js (hunk 1 above)
Comment (ottumm) [inline lib/rules/disallow-tabs.js]: Yes, gladly. Did you have something in mind?

Comment (ottumm) [inline lib/rules/disallow-tabs.js]: Maybe "Tab characters are not allowed"?

**Comment by markelog** [comment]
> yeah, would it be better to extend validateIndentation to handle this? My use case is to completely disallow tabs.

I think it is better to have a different rule for it.

>  I was finding that validateIndentation wasn't doing that and in fact was exhibiting problems when tabs were present.

What problems?

File: lib/rules/disallow-tabs.js (hunk 1 above)
Comment (markelog) [inline lib/rules/disallow-tabs.js]: I meant put a column number there,

**Comment by ottumm** [comment]
I filed issue #1762 to track the validateIndentation issue I'm seeing.

File: lib/rules/disallow-tabs.js (hunk 1 above)
Comment (ottumm) [inline lib/rules/disallow-tabs.js]: Ah, gotcha, ok.

Comment (ottumm) [inline lib/rules/disallow-tabs.js]: https://github.com/ottumm/node-jscs/commit/e0dad2ebd3c1089410e950febd76d2a7c17ea8af addresses this.

**Comment by markelog** [comment]
Landed!

**Comment by ottumm** [comment]
Awesome, thanks!

**Pull request closed without merging**